src/seedu/addressbook/ui/TextUi.java

```diff
@@ -164,5 +165,21 @@ private static String getIndexedListForViewing(List<String> listItems) {
     private static String getIndexedListItem(int visibleIndex, String listItem) {
         return String.format(MESSAGE_INDEXED_LIST_ITEM, visibleIndex, listItem);
     }
+    
+    public String getPersonPrintableString(ReadOnlyPerson p) {
+    	return getPrintableString(p.getName(), p.getPhone(), p.getEmail(), p.getAddress());
+    }
+    
+    /**
+     * Returns a concatenated version of the printable strings of each object.
+     */
+    String getPrintableString(Printable... printables) {
+    	final StringBuilder builder = new StringBuilder();
+    	for (Printable printable : printables) {
+    		builder.append(printable.getPrintableString());
+    		builder.append(" ");
```

Comment:
I think this result in an extra " " at the end. Take a look @ StringJoiner or String.join as mentioned in tutorial :)